presto-hive/src/test/java/com/facebook/presto/hive/TestBackgroundSplitLoader.java

```diff
@@ -0,0 +1,320 @@
+
+public class TestBackgroundSplitLoader
+{
+    private static final int MAX_BUCKETS = 2;
+
+    private static final String SAMPLE_PATH = "hdfs://srv/test/000000_0";
+    private static final String SAMPLE_PATH_FILTERED = "hdfs://srv/test/000000_1";
+    private static final String TEST_CONNECTOR_ID = "test_connector";
+    private static final String TEST_COLUMN_NAME = "testcolumn";
+    private static final String TEST_DATABASE = "testdb";
+    private static final String TEST_FILE_FORMAT = "org.apache.hadoop.hive.ql.io.RCFileInputFormat";
+    private static final String TEST_PARTITION_COLUMN_NAME = "partitionColumn";
+    private static final String TEST_SCHEMA = "testschema";
+    private static final String TEST_TABLE_LOCATION = "hdfs://srv/test";
+    private static final String TEST_TABLE_MOCK_OWNER = "testOwner";
+    private static final String TEST_TABLE_NAME = "table";
+
+    private final MockExecutor mockExecutor;
+    private final Path returnedPath;
+    private final Path filteredPath;
+
+    public TestBackgroundSplitLoader()
+    {
+        this.mockExecutor = new MockExecutor();
+        this.returnedPath = new Path(SAMPLE_PATH);
+        this.filteredPath = new Path(SAMPLE_PATH_FILTERED);
+    }
+
+    @Test
+    public void testNoPathFilter() throws Exception
+    {
+        final BackgroundHiveSplitLoader backgroundHiveSplitLoader = getBackgroundHiveSplitLoader(getFiles(), TupleDomain.none());
+        final HiveSplitSource hiveSplitSource = new HiveSplitSource(1,
+                backgroundHiveSplitLoader,
+                mockExecutor);
+
+        backgroundHiveSplitLoader.start(hiveSplitSource);
+        assertEquals(2, hiveSplitSource.getOutstandingSplitCount());
+    }
+
+    @Test
+    public void testPathFilter() throws Exception
+    {
+        final BackgroundHiveSplitLoader backgroundHiveSplitLoader = getBackgroundHiveSplitLoader(getFiles(),
+                getPathTupleDomain(returnedPath));
+        final HiveSplitSource hiveSplitSource = new HiveSplitSource(1,
+                backgroundHiveSplitLoader,
+                mockExecutor);
+
+        backgroundHiveSplitLoader.start(hiveSplitSource);
+        assertEquals(1, hiveSplitSource.getOutstandingSplitCount());
+
+        final List<ConnectorSplit> connectorSplits = hiveSplitSource.getNextBatch(1).get();
+        assertEquals(1, connectorSplits.size());
+
+        final HiveSplit connectorSplit = (HiveSplit) connectorSplits.get(0);
+        assertEquals(returnedPath.toString(), connectorSplit.getPath());
+    }
+
+    @Test
+    public void testPathFilterBucketedPartitionedTable() throws Exception
+    {
+        final List<Column> partitionColumns = new ArrayList();
+        partitionColumns.add(new Column(TEST_PARTITION_COLUMN_NAME, HIVE_INT, Optional.empty()));
+
+        final Table partitionedTable = getTable(partitionColumns);
+
+        final BackgroundHiveSplitLoader backgroundHiveSplitLoader = getBackgroundHiveSplitLoader(getFiles(),
```

Comment:
Fixed for most of the cases, in a couple of instances, left them for readability.